src/librustdoc/html/sources.rs

```diff
@@ -54,12 +54,16 @@ impl DocFolder for SourceCollector<'_, '_> {
             self.scx.include_sources = match self.emit_source(&filename) {
                 Ok(()) => true,
                 Err(e) => {
-                    println!(
-                        "warning: source code was requested to be rendered, \
-                         but processing `{}` had an error: {}",
-                        filename, e
+                    let mut warning = self.scx.tcx.sess.struct_span_warn(
+                        item.source.span(),
+                        &format!(
+                            "source code was requested to be rendered, \
+                             but an error occurred: {}",
+                            e
+                        ),
```

Comment:
This seems a little verbose.
```suggestion
                        &format!(
                            "failed to render source code: {}",
                            e
                        ),
```
(you may need to run `x.py fmt` after)

Reply:
I agree about the message change, but the span shown for the diagnostic should already show the file that couldn't be rendered. If you prefer we say it in the error message, I can do that, but wanted to let you know that we should already show it.

Reply:
I ended up adding the filename after all.

Reply:
> I agree about the message change, but the span shown for the diagnostic should already show the file that couldn't be rendered. If you prefer we say it in the error message, I can do that, but wanted to let you know that we should already show it.

It shows the original file, but not the file rustdoc tried to create, which is usually more relevant.